include/natalie/global_env.hpp

```diff
@@ -139,6 +141,12 @@ class GlobalEnv : public Cell {
 
     ClassObject *m_Encodings[EncodingCount];
 
+    struct span {
```

Comment:
I think we should call this something more descriptive like `StringInternSpan` or some such. Your call on the name, but as is, fully-namespaced, this one is `Natalie::GlobalEnv::span`, which doesn't communicate what it's for.

Reply:
I think https://github.com/seven1m/tm/pull/11 is the more generic solution here, by extracting a generic Span class that is totally disjoint from StringObject.